### DISCUSSION TITLE
[12일차-윤경]DOM이란 무엇일까요

### DISCUSSION THREAD

**Discussion by skyhanull** [zzangCS/zzangInterview#42] [프론트 면접 질문]
DOM에 대해 설명해주세요

**Comment by smarfy99** [comment]
DOM은 Document object model로 html이나 xml에 접근하기 위한 일종의 인터페이스.
문서 내의 모든 요소를 정의하고, 각각의 요소에 접근하는 방법을 제공한다.
- HTML 요소들의 구조화된 표현이다. 
- JS로 DOM을 수정할 수 있으며 이는 HTML문서의 내용을 직접 변경하는 것이 아니다.
- 가상 요소를 포함하지 않는다.
- 보이지 않는 요소를 포함한다. (display : none)

**Comment by skyhanull** [comment]
DOM (Document Object Model)의 약자로 웹페이지를 프로그래밍적으로 제어할 수 있게 해주는 객체모델 이다. 최상위 인터페이스로 Node가 존재합니다.트리 자료구조로 구축되기 때문에 DOM Tree라고 부르기도 합니다.
자바스크립트는 DOM을 조작할 수 있는 대표적인 언어입니다. 현재 웹 브라우저에서 DOM을 조작할 수 있는 언어는 자바스크립트 뿐이지만, 다른 프로그래밍 언어로도 DOM을 구현할 수 있습니다.

**Comment by ez1n** [comment]
DOM (Document Object Model)

⇒ HTML과 XML 문서의 프로그래밍 인터페이스 (nodes, objects로 문서 표현)
⇒ HTML은 브라우저에서 실행도리 수 있게끔 DOM Tree로 파싱되고 이를 바탕으로 렌더링한다.
⇒ 문서의 구조화된 표현을 제공하며 프로그래밍 언어가 DOM 구조에 접근할 수 있는 방법을 제공하여 문서 구조, 스타일, 내용 등을 변경할 수 있게 돕는 역할
⇒ 가상 요소는 포함하지 않으며 보이지 않는 요소는 포함
